WEB-INTEGRATION.md

```diff
@@ -301,47 +306,44 @@ This is also the case for worklets, where the registering API (e.g.
 for audio worklets [\[WEBAUDIO\]](https://webaudio.github.io/web-audio-api/), or
 [`registerPaint()`](https://drafts.css-houdini.org/css-paint-api-1/#dom-paintworkletglobalscope-registerpaint)
 for paint worklets [\[CSS-PAINT-API\]](https://drafts.css-houdini.org/css-paint-api-1/))
-is the registration context, and the causal context is either empty or
-unobservable (since `AsyncContext.Variable`s from outside the worklet cannot
-cross its boundary, even if they happen to live in the same agent/thread).
-Therefore, the registration context should be used.
+is the "registration-time context", and the context of the JavaScript code that
+actually caused the callback to run is _always_ either empty or unobservable
+(since `AsyncContext.Variable`s from outside the worklet cannot cross its boundary,
+even if they happen to live in the same agent/thread).
 
 For action registrations where the action often originates from userland JS
-code, the causal context should be used instead. The main case for this is
-custom elements, where the lifecycle callbacks are almost always triggered
+code, we should instead follow the main principle of propagating the context from
+the point that started the work that eventually triggered the callback. The main
+case for this is custom elements, where the lifecycle callbacks are almost always triggered
 synchronously by a call from userland JS to an API annotated with
 [`[CEReactions]`](https://html.spec.whatwg.org/multipage/custom-elements.html#cereactions).
-However, there are cases where this is not the case:
+However, there are cases where this is not the case and falls back to the empty
+context where each `AsyncContext.Variable` is set to its initial value:
 
 - If a custom element is contained inside a `<div contenteditable>`, the user
   could remove the element from the tree as part of editing, which would queue a
-  microtask to call its `disconnectedCallback` hook. In this case, there would
-  be no causal context, and each `AsyncContext.Variable` would be set to its
-  initial value.
+  microtask to call its `disconnectedCallback` hook.
 - A user clicking a form reset when a form-associated custom element is in the
   form would queue a microtask to call its `formResetCallback` lifecycle hook,
-  and there would not be a casual context. However, if the `click()` method is
-  called from JS instead, since that method doesn't have the `[CEReactions]`
-  annotation, it would also call that lifecycle hook in a microtask, rather than
-  synchronously. In that case, the causal context would be the one active when
-  `.click()` was called.
+  and there would not be a causal context.
+
+<!-- Is this still true? -->
```

Comment:
Yeah, if this is a user click, the task that handles the form reset would be triggered by browser-internal code, and that empty context would propagate to the microtask that calls the lifecycle hook.